fe/lib/components/SalaryDetails/v2/README.md

```diff
@@ -0,0 +1,98 @@
+# SalaryDetailsV2
+
+A Salary Details component that contains information on the salary type, range and description.
+
+## Installation
+
+```shell
+$ yarn add wingman-fe
+```
+
+### Validation
+
+The Salary Details component has some basic UI validation rules:
+
+- The minimum pay must be greater than 0.
+- The maximum pay must be greater than the minimum pay.
+
+### Usage
+
+```javascript
+import React, { useState } from 'react';
+import { SalaryDetailsV2 } from 'wingman-fe';
+
+const buildRemunerationPackageInput = (remuneration) => ({
+  basisCode: remuneration.basisCode,
+  ranges: [{
+    intervalCode: remuneration.intervalCode,
+    minimumAmount: {
+      currency: remuneration.currency,
+      value: Number(remuneration.minimumAmount),
+    },
+    maximumAmount: remuneration.maximumAmount ?? {
+      currency: remuneration.currency,
+      value: Number(remuneration.maximumAmount),
+    },
+  }],
+  descriptions: remuneration.description ? [remuneration.description] : [],
+})
+
+const PostingForm = () => {
+  const [remuneration, setRemuneration] = useState({
+    basisCode: '',
+    currency: 'AUD',
+    description: '',
+    intervalCode: '',
+    maximumAmount: null,
+    minimumAmount: '',
+    payType: '',
+  });
+
+  const onBlur = (item) => {
```

Comment:
Thinking about this discriminated union logic again, I'm thinking we can keep this as is. I like the way this provides an easy way to act on specific changes to certain data in the form and is flexible to whether you want to store the `remuneration` state as one object or in individual states. If we package the `onBlur`'s return into a single object, the example would be a bit cleaner but then we lose out on that flexibility.